apps/sv/src/main/scala/org/lfdecentralizedtrust/splice/sv/store/db/DbSvDsoStore.scala

```diff
@@ -526,15 +537,18 @@ class DbSvDsoStore(
                """.as[(Long, Array[ContractId[ValidatorRewardCoupon]])],
             opName,
           )
-      } yield applyLimit(opName, totalCouponsLimit, result).map { case (round, batch) =>
-        RoundBatch(
-          round,
-          batch
-            .take(totalCouponsLimit.limit)
-            .map(cid => companionClass.toContractId(companion, cid.contractId))
-            .toSeq,
-        )
-      }
+      } yield applyLimit(opName, totalCouponsLimit, result)
+        .map { case (round, batch) =>
+          RoundBatch(
```

Comment:
Doesn't this have exactly the problem I mentioned where you can end up fetching only an empty list why I wanted to move it into the DB queries? Picking some extreme values to make the example clearer:

Let's say totalCouponsLimit = 1. Your DB query fetches one batch. Let.s assume that one batch only contains ignored contracts, you then filter out the ignored contracts so now you have no batch. And you just repeat that over and over.

Reply:
The limit is for the number of rounds we're fetching. Indeed this is a valid scenario. We could in theory then just increase the batch size and we should fix it as no new contracts that have this issue should be created. 

Let me just move the filtering into the query though, only issue I'm seeing is if we're somehow exceeding some limits of how many params can we send in the query.

Reply:
That sounds like a better solution, easier to source the data as well. Let me just quickly change to ignore party